packages/amplify-category-hosting/index.js

```diff
@@ -58,9 +59,17 @@ async function configure(context) {
 
 function publish(context, service, args) {
   const { enabledServices } = categoryManager.getCategoryStatus(context);
-
   if (enabledServices.length > 0) {
     if (enabledServices.includes(service)) {
+      if (service === 'S3AndCloudFront' && context.exeInfo?.amplifyMeta.hosting.S3AndCloudFront.output) {
+        if (context.exeInfo.amplifyMeta.hosting.S3AndCloudFront.output['CloudFrontSecureURL'] === undefined) {
+          throw new AmplifyError('ProjectPublishError', {
+            message: 'You are trying to host an application without a CloudFront distribution. This is not supported on S3 by default.',
+            details: 'You will need to deploy your application to PROD or use this workaround for non-public dev apps only.',
+            link: 'https://github.com/aws-amplify/amplify-cli/issues/12503',
```

Comment:
This is for the case when someone has added hosting from an S3 bucket with a CloudFront distribution on a previous version of CLI. I need feedback on the messaging here.

Reply:
I am going to add a comment to document what this is doing here